scheds/rust/scx_rusty/src/main.rs

```diff
@@ -262,10 +263,25 @@ impl<'a> Scheduler<'a> {
         skel.rodata_mut().nr_doms = domains.nr_doms() as u32;
         skel.rodata_mut().nr_cpus = top.nr_cpus() as u32;
 
+        let mut warned = false;
         for cpu in 0..top.nr_cpus() {
-            let dom_id = domains.cpu_dom_id(&cpu).unwrap();
-            skel.rodata_mut().cpu_dom_id_map[cpu] =
-                dom_id.clone().try_into().expect("Domain ID could not fit into 32 bits");
+            if let Some(dom_id) = domains.cpu_dom_id(&cpu) {
+                skel.rodata_mut().cpu_dom_id_map[cpu] =
+                    dom_id.clone().try_into().expect("Domain ID could not fit into 32 bits");
+            } else {
+                // As described in
+                // https://bugzilla.kernel.org/show_bug.cgi?id=218109, some
+                // chips may incorrectly report disabled CPUs as offline. It
+                // doesn't appear that this is going to be corrected anytime
+                // soon, so just put the CPUs in domain 0 so we can get past
+                // startup.
+                if !warned {
+                    warn!("Found a CPU without a domain, putting in domain 0. May experience suboptimal scheduling.");
+                    warn!("Please see https://bugzilla.kernel.org/show_bug.cgi?id=218109 for more details");
```

Comment:
Given how widespread this is, I'm not sure warning & possibly causing suboptimal behavior can be a solution. If our plan to handle hotplugs is by reloading scheduler, maybe the topology code can always just report online cpus and we can ignore the discrepancies between possible and online CPUs in the schedulers?